atomic_defi_design/Dex/Components/ZcashParamsModal.qml

```diff
@@ -0,0 +1,122 @@
+// Qt Imports
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+
+// Project Imports
+import "../Constants" as Dex
+import "../Components" as Dex
+import App 1.0
+
+Dex.MultipageModal
+{
+    id: root
+    readonly property var coins: API.app.zcash_params.get_enable_after_download()
+    width: 750
+    closePolicy: Popup.NoAutoClose
+
+    // Inside modal
+    Dex.MultipageModalContent
+    {
+        titleText: qsTr("%1 Activation Failed!").arg(coins.join(' / '))
+
+        Dex.DefaultText
+        {
+            Layout.fillWidth: true
+            text: qsTr("To activate ZHTLC coins, you need to download the Zcash Params.\nThis might take a few minutes...")
+        }
+
+        HorizontalLine
+        {
+            Layout.topMargin: 8
+            Layout.bottomMargin: 8
+            Layout.fillWidth: true
+        }
+
+        ColumnLayout
+        {
+            Dex.DefaultProgressBar
+            {
+                id: sapling_output_params
+                label.text: "sapling-output.params"
+                bar_width_pct: 0
+                pct_value.text: "0.00 %"
+            }
+            Dex.DefaultProgressBar
+            {
+                id: sapling_spend_params
+                label.text: "sapling-spend.params"
+                bar_width_pct: 0
+                pct_value.text: "0.00 %"
+            }
+        }
+
+        HorizontalLine
+        {
+            Layout.topMargin: 8
+            Layout.bottomMargin: 8
+            Layout.fillWidth: true
+        }
+
+        footer:
+        [
+            Item { Layout.fillWidth: true },
+            Dex.DefaultButton
+            {
+                id: download_button
+                text: qsTr("Download params & enable coins")
+                enabled: !sapling_output_params.bar_width_pct > 0 || !sapling_spend_params.bar_width_pct > 0
+                onClicked: {
+                    download_button.enabled = false
+                    Dex.API.app.zcash_params.download_zcash_params()
+                }
+            },
+            Item { Layout.fillWidth: true },
+            Dex.DefaultButton
+            {
+                text: qsTr("More Info")
+                onClicked: Qt.openUrlExternally("https://www.coinbureau.com/education/zcash-ceremony/")
+            },
+            Item { Layout.fillWidth: true },
+            Dex.DefaultButton
+            {
+                text: qsTr("Close")
+                onClicked: close()
+            },
+            Item { Layout.fillWidth: true }
+        ]
+
+        Connections
+        {
+            target: Dex.API.app.zcash_params
+            function onCombinedDownloadStatusChanged()
```

Comment:
This implementation is slow and error prone, it should be done in C++ where there is strong type safety: put at least a todo for next releases